### ISSUE TITLE
Unable to use oadm on guest VM

### ISSUE THREAD

**Issue by hferentschik** [projectatomic/adb-utils#72]
_openshift_provision_ does copy the following binaries:

```
binaries=(openshift oc oadm)
for n in ${binaries[@]}; do
  echo "[INFO] Copying ${n} binary to VM"
  docker cp openshift:/usr/bin/${n} /usr/bin/${n}
done
```

However with (at least) OSE 3.1 and OSE 3.2 oadm is really a symbolic link to openshift within the container. Downloading all three binaries like this, seems to break the links. Trying to execute oadm results into the following error:

```
/usr/bin/oadm
-bash: /usr/bin/oadm: Too many levels of symbolic links
```

Not quite sure why this had not popped up earlier.

I guess we just need to copy `oc` and `openshift` and then create the symbolic link ourselves.

**Comment by hferentschik** [comment]
Can we add focus:devex label? Also if this is reproducible, it's a blocker

**Comment by hferentschik** [comment]
FYI - https://github.com/redhat-developer-tooling/openshift-vagrant/commit/ddd3dba273a40ee2093f7c54fb7bde1f63e60236

**Comment by praveenkumar** [comment]
@hferentschik  I didn't face this issue but give latest box to another try.

**Comment by praveenkumar** [comment]
@hferentschik I gave a spin to latest box and didn't able to reproduce this issue. Can you tell if I have to follow some specific steps to reproduce it?

I also tried `vagrant halt`, `vagrant provision` but still working as expected.

```
$ oadm -h
Administrative Commands

Commands for managing a cluster are exposed here. Many administrative
actions involve interaction with the command-line client as well.

Basic Commands:
  new-project                       Create a new project
  policy                            Manage policy
  groups                            Manage groups

Install Commands:
  router                            Install a router
  ipfailover                        Install an IP failover group to a set of nodes
  registry                          Install the integrated Docker registry

Maintenance Commands:
  build-chain                       Output the inputs and dependencies of your builds
  manage-node                       Manage nodes - list pods, evacuate, or mark ready
  prune                             Remove older versions of resources from the server

Settings Commands:
  config                            Change configuration files for the client
  create-kubeconfig                 Create a basic .kubeconfig file from client certs
  create-api-client-config          Create a config file for connecting to the server as a user

Advanced Commands:
  pod-network                       Manage pod network
  create-bootstrap-project-template Create a bootstrap project template
  create-bootstrap-policy-file      Create the default bootstrap policy
  create-login-template             Create a login template
  overwrite-policy                  Reset the policy to the default values
  create-node-config                Create a configuration bundle for a node
  ca                                Manage certificates and keys

Other Commands:
  version                           Display version
  help                              Help about any command

Use "oadm help <command>" for more information about a given command.
Use "oadm options" for a list of global command-line options (applies to all commands).
```

**Comment by hferentschik** [comment]
Yeah, I am not so sure either atm. I have seen the problem on my VM and other team members have reported it as well. But I am also not quite sure how it happens. Somehow it has to related to the fact that oadm is just a link to openshift, but I also cannot reproduce this repeatably atm. I try to do some more investigations.

**Comment by praveenkumar** [comment]
@hferentschik  I found root case of this issue and we have resolved it in our #75 PR. Issue was `oadm` is symlink to `openshift` and provision script again try to create same symlink which case _Too many levels of symbolic link_ issue.

```
* First run of openshift provision *
# cd /usr/bin
# find -L ./ -mindepth 15

* Subsequent run of openshift provision*
# cd /usr/bin
# find -L ./ -mindepth 15
find: ‘./openshift’: Too many levels of symbolic links
find: ‘./oadm’: Too many levels of symbolic links
```

**Comment by hferentschik** [comment]
> I found root case of this issue and we have resolved it in our #75 PR. Issue was oadm is symlink to openshift and provision script again try to create same symlink which case Too many levels of symbolic link issue

Nice one :-) Good catch. That's why it worked the first time, but then after restarting VM or service you would get into trouble. Good that we caught this one.

**Comment by hferentschik** [comment]
Personally I would separate the pull request. Aka one for for issue #74 and one for #72, but that's just a minor thing.

**Comment by praveenkumar** [comment]
Fixed with #75
